server/UserRegistration.java

```diff
@@ -0,0 +1,357 @@
+package newbank.server;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Scanner;
+
+public class UserRegistration {
+
+    private static Scanner sc = new Scanner(System.in);
+        UserRegistration ur = UserRegistration.newUser();
+    public UserRegistration() throws SQLException {
+    }
+
+    public static UserRegistration newUser() throws SQLException {
+        // set all methods to return values - to then push to db.
+        String UserFullname = UserRegistration.getNameFromUser();
+        String UserDob = UserRegistration.getDateFromUser();
+        String UserId = UserRegistration.getNewUserId();
+        String UserAccountNumber = UserRegistration.getNewUserAccountNumber();
+        String UserSortcode = UserRegistration.getNewUserSortCode();
+        String UserPassword = UserRegistration.getNewUserInputNewPassword();
+        Integer UserBalance = Integer.valueOf(UserRegistration.getNewUserBalance());
+        String UserSecretQuestion = UserRegistration.getUserSecretQuestion();
+        Boolean UserCheckHuman = UserRegistration.getUserCheckHuman();
+
+
+        // All already scanned by scanner then passed as values to 'String addNewUser'
+        Connection conn = null;
+        try {
+            String url = "jdbc:sqlite:identifier.sqlite";
+            conn = DriverManager.getConnection(url);
+            // Will allow for the database connection.
+            String addNewUser = "INSERT INTO userDatabase(full_name, dob, username, account_number, sortcode, password, balance, secret_answer) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+
+            // New values passed to db.
+            PreparedStatement updateUserdb = conn.prepareStatement(addNewUser);
+
+            updateUserdb.setString(1, UserFullname);
+            updateUserdb.setString(2, UserDob);
+            updateUserdb.setString(3, UserId);
+            updateUserdb.setString(4, UserAccountNumber);
+            updateUserdb.setString(5, UserSortcode);
+            updateUserdb.setString(6, UserPassword);
+            updateUserdb.setInt(7, UserBalance);
+            updateUserdb.setString(8, UserSecretQuestion);
+            updateUserdb.executeUpdate();
+            conn.commit();
+            // The db will now push the output.
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        } finally {
+            try {
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException ex) {
+                System.out.println(ex.getMessage());
+            }
+        }
+        return null;
+    }
+
+    private static String getNameFromUser() {
```

Comment:
Why are all these methods `static`? As far as I can tell, there's no need for them to be, `static` keyword could be removed